sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala

```diff
@@ -101,6 +101,17 @@ case class CreateViewStatement(
     replace: Boolean,
     viewType: ViewType) extends ParsedStatement
 
+/**
+ * A CREATE TEMP VIEW USING statement, as parsed from SQL.
+ */
+case class CreateTempViewUsingStatement(
```

Comment:
I'm not sure we need it. We use statements for commands that have both v1 and v2 versions, and we need to pick one in the analyzer.

temp view is managed by Spark itself, so we only have one command for CREATE TEMP VIEW USING. We should just reuse the existing `CreateTempViewUsing` command.

Reply:
Thanks for the explanation. So I guess we don't need any of the changes except the one removing ```visitCreateTable``` from ```SparkSqlParser```?

Reply:
We can move `CreateTempViewUsing` to catalyst and simply the code. Removing the old CREATE EXTERNAL TABLE USING is another topic and should be discussed more as it's a breaking change.

Reply:
`CREATE EXTERNAL TABLE ... USING` is disallowed for now. So I think you mean`CREATE TEMPORARY TABLE ... USING`? @cloud-fan

Reply:
So `CreateTempViewUsing` always uses session catalog if a catalog is provided?

Reply:
ah yes, CREATE TEMPORARY TABLE ... USING

`CreateTempViewUsing` doesn't need catalog at all. It needs `SessionCatalog` now because somehow we manage temp views by `SessionCatalog`, but we may refactor it in the future.

Reply:
hmm, `CreateTempViewUsing` is an execution command and depends on few things in sql core. Seems we can not move it simply.

Reply:
then there is not much we can do...

Reply:
I will close this PR. Thanks a lot for your help! @viirya @cloud-fan